### PR TITLE
[SPMD] Replace nn.Linear

### PR DESCRIPTION
**Pull request by alanwaketan**
Summary:
This pull request replaces the default nn.Linear with our patched version that doesn't flatten the high dimensional tensors.

Test Plan:
Tested on a V4-8.

### CODE REVIEW

**Review by jonb377 — APPROVED** [review]
LGTM

File: examples/pytorch/language-modeling/run_clm.py [hunk 1]
```diff
@@ -543,6 +543,10 @@ def get_mesh(ici_mesh_shape, dcn_mesh_shape=None):
       else:
         return xs.HybridMesh(ici_mesh_shape=ici_mesh_shape, dcn_mesh_shape=dcn_mesh_shape)
 
+    # Replace the linear layer
+    from torch_xla.distributed.fsdp.utils import apply_xla_patch_to_nn_linear
+    model = apply_xla_patch_to_nn_linear(model, xs.xla_patched_nn_linear_forward)
```
Comment (jonb377) [inline examples/pytorch/language-modeling/run_clm.py:548]: What's the difference between `xs.xla_patched_nn_linear_forward` and the default patch method?

Comment (jonb377) [reply examples/pytorch/language-modeling/run_clm.py:548]: Oh I see - the patch classes are different.

**Comment by alanwaketan** [comment]
Thanks, Jon.

**Pull request merged**